setup_cares.py

```diff
@@ -98,30 +94,20 @@ def build_extensions(self):
         build_ext.build_extensions(self)
 
     def build_cares(self):
-        #self.debug_mode =  bool(self.debug) or hasattr(sys, 'gettotalrefcount')
-        win32_msvc = self.compiler.compiler_type == 'msvc'
-        def build():
-            cflags = '-fPIC'
-            env = os.environ.copy()
-            env['CFLAGS'] = ' '.join(x for x in (cflags, env.get('CFLAGS', None)) if x)
-            log.info('Building c-ares...')
-            if win32_msvc:
-                # pass the found compiler to Makefile.msvc
-                env['CC'] = self.compiler.cc
-                env['AR'] = self.compiler.lib
-                exec_process('cmd.exe /C vcbuild.bat', cwd=self.cares_dir, env=env, shell=True, silent=False)
-            else:
-                exec_make(['libcares.a'], cwd=self.cares_dir, env=env, silent=False)
-        def clean():
-            if win32_msvc:
-                exec_process('cmd.exe /C vcbuild.bat clean', cwd=self.cares_dir, shell=True)
-            else:
-                exec_make(['clean'], cwd=self.cares_dir)
-        if self.cares_clean_compile:
-            clean()
-        if not os.path.exists(self.cares_lib):
-            log.info('c-ares needs to be compiled.')
-            build()
+        self.compiler.define_macro('HAVE_CONFIG_H', 1)
+        if sys.platform == 'win32':
+            global cares_sources
+            cares_sources += ['deps/c-ares/src/windows_port.c',
+                              'deps/c-ares/src/ares_platform.c']
+            self.compiler.add_include_dir(os.path.join(self.cares_dir, 'src/config_win32'))
+            #-D_WIN32_WINNT=0x0600 
+
         else:
-            log.info('No need to build c-ares.')
+            self.compiler.define_macro('_LARGEFILE_SOURCE', 1)
+            self.compiler.define_macro('_FILE_OFFSET_BITS', 64)
+
+            if sys.platform.startswith('linux'):
+                self.compiler.add_include_dir(os.path.join(self.cares_dir, 'src/config_linux'))
```

Comment:
yes, I can add it according to the original Makefile . but I have no osx to verify it.